zerver/tests/test_auth_backends.py

```diff
@@ -328,82 +292,104 @@ def test_remote_user_backend_sso_append_domain(self) -> None:
+                        'method': httpretty.GET,
+                        'status': 200,
+                        'body': json.dumps(dict(email=user.email, name=user.full_name, email_verified=True)),
+                    },
+                ],
+                'backend': GoogleAuthBackend,
+            },
+            'github': {
+                'urls': [
+                    {
+                        'url': "https://github.com/login/oauth/access_token",
+                        'method': httpretty.POST,
+                        'status': 200,
+                        'body': json.dumps(token_data_dict),
+                    },
+                    {
+                        'url': "https://api.github.com/user",
+                        'method': httpretty.GET,
+                        'status': 200,
+                        'body': json.dumps(dict(email=user.email, name=user.full_name)),
+                    },
+                    {
+                        'url': "https://api.github.com/user/emails",
+                        'method': httpretty.GET,
+                        'status': 200,
+                        'body': json.dumps(github_email_data),
+                    },
+                ],
+                'backend': GitHubAuthBackend,
+            }
+        }   # type: Dict[str, Any]
 
         def patched_authenticate(*args: Any, **kwargs: Any) -> Any:
             if 'subdomain' in kwargs:
                 backend.strategy.session_set("subdomain", kwargs["subdomain"])
                 del kwargs['subdomain']
             result = orig_authenticate(backend, *args, **kwargs)
             return result
-        backend.authenticate = patched_authenticate
-        good_kwargs = dict(backend=backend, strategy=backend.strategy,
-                           storage=backend.strategy.storage,
-                           response=token_data_dict,
-                           subdomain='zulip')
-        bad_kwargs = dict(subdomain='acme')
-        with mock.patch('zerver.views.auth.redirect_and_log_into_subdomain',
-                        return_value=user):
-            self.verify_backend(backend,
-                                good_kwargs=good_kwargs,
-                                bad_kwargs=bad_kwargs)
-            bad_kwargs['subdomain'] = "zephyr"
-            self.verify_backend(backend,
-                                good_kwargs=good_kwargs,
-                                bad_kwargs=bad_kwargs)
-        backend.authenticate = orig_authenticate
-        httpretty.disable()
-
-class ResponseMock:
-    def __init__(self, status_code: int, data: Any) -> None:
-        self.status_code = status_code
-        self.data = data
-
-    def json(self) -> str:
-        return self.data
 
-    @property
-    def text(self) -> str:
-        return "Response text"
+        for backend_name in backends_to_test:
+            httpretty.enable(allow_net_connect=False)
```

Comment:
Enabling this option is a nice improvement!  I added it to the other user we have of this library in 73655a6176967ffd1a11299e15c5f5b6603e017b.  Though reading https://httpretty.readthedocs.io/en/latest/api.html#enable, I wonder if even better would be to make it a decorator on the test / a with block?